data_engineering/pipeline_runner.py

```diff
@@ -59,17 +45,15 @@ def run():
     }
 
     pipeline_io.create_output_dir()
-    df = pipeline_io.read_file(TRAIN)
 
-    transformed_df = transform(df)
-
-    raw_values_df = get_raw_values(df)
-
-    pipeline_io.save_file(transformed_df, output_files[f"ratio_{TRAIN}"])
+    df = pipeline_io.read_file(TRAIN)
+    transformed_ratio_df = transform_ratio(df)
+    transformed_raw_df = transform_raw(df)
```

Comment:
I can see most of the code is duplicate now... it's ok, only if you drop duplicate here, you can't concat two files together later. 
One solution being concat these 2 files first then have a third transform function to process it. Then pipeline_runner will produce 3 files

Reply:
yeah... the `transform_ratio` and `transform_raw` are quite similar... i might actually do something like this: keep `transform_ratio`, remove `transform_raw` then in `run` just apply `get_raw_values` to the `transformed_ratio_df`, then we can concat the 2 files. Is this better?